pySearch.py

```diff
@@ -44,7 +52,11 @@ def buildLink(self):
     #end of link building
 
     def openBrowser(self):
-        webbrowser.open_new_tab(self.url)
+        try:
+            browser = webbrowser.get(args.browser[0])
```

Comment:
Great first start but this is unsafe. There is no default state if the browser argument isn't supplied and overall I would avoid handling the browser argument in `openBrowser` as it makes the function harder to test with a test suite. Instead could you create a handler for browser arguments(`setBrowser` perhaps)? 

If the user doesn't input a browser then you don't need to run `webbrowser.get`. As it is right now your solution makes the `-b` argument mandatory.

If possible I would like to see this handler implement a bit of logic to allow aliases for the different browsers (so I can call google chrome by typing Chrome, google or some other shorthand). Something like this could be achieved with a [regular expression](https://docs.python.org/3/library/re.html). 

If I wanted the `firefox` browser the regular expression `[Ff]irefox` would allow caps for firefox or using `([Gg]oogle)|([Cc]hrome)|([Gg]oogle [Cc]hrome)` would allow the `google` alias to map to the preregistered `chrome` browser. This would mitigate some degree of user error and I think it would be a great quality of life improvement.

Reply:
I will continue to develop this with the specified functionalities, thanks for the recommendations!